cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/ConfigurationFinder.java

```diff
@@ -89,49 +115,46 @@ private static Properties readConfigrationFile(Path cobigenConfigFile) {
   }
 
   /**
-   * This is a helper method to find templates in cobigen home
+   * This is a helper method to find templates or template sets in CobiGen home
    *
-   * @param home cobigen configuration home directory
-   * @return templates location if found, otherwise null
+   * @param home CobiGen configuration home directory
+   * @return templates / template sets location if found, otherwise null
    */
   private static URI findTemplates(Path home) {
 
     Path templatesPath = CobiGenPaths.getTemplatesFolderPath(home);
     Path templatesFolderPath = templatesPath.resolve(ConfigurationConstants.COBIGEN_TEMPLATES);
 
-    // 1. use Cobigen_Templates folder
+    // 1. use old Cobigen_Templates folder
     if (Files.exists(templatesFolderPath)) {
       return templatesFolderPath.toUri();
     }
 
-    // 2. use template jar
-    Path jarPath = getTemplateJar(templatesPath);
-    if (jarPath != null) {
-      return jarPath.toUri();
+    // 2 create/use new template sets folder
+    Path templateSetsFolderPath = CobiGenPaths.getTemplateSetsFolderPath(home);
+
+    Path templateSetsAdaptedFolderPath = templateSetsFolderPath.resolve(ConfigurationConstants.ADAPTED_FOLDER);
+
+    // 3 check adapted folder
+    if (Files.exists(templateSetsAdaptedFolderPath)) {
+      return templateSetsFolderPath.toUri();
+    }
+
+    Path templateSetsDownloadedFolderPath = templateSetsFolderPath.resolve(ConfigurationConstants.DOWNLOADED_FOLDER);
+
+    // 4 check downloaded folder
+    if (Files.exists(templateSetsDownloadedFolderPath)) {
+      return templateSetsFolderPath.toUri();
```

Comment:
Couldn't check 3 and 4 be placed in one single if clause?

Reply:
Combined.